pkg/docker/docker.go

```diff
@@ -24,8 +25,23 @@ type Client interface {
 
 var _ Client = &docker.Client{}
 
-func Tag(registry, image, tag string) string {
-	return fmt.Sprintf("%s/%s:%s", registry, image, tag)
+func Tag(registry, image, tag string, eout io.Writer) string {
+	slug := SlugifyTag(tag)
+	if slug != tag {
+		_, _ = fmt.Fprint(eout, tml.Sprintf("<yellow>Warning: tag was changed from '%s' to '%s' due to Dockers rules.</yellow>", tag, slug))
+	}
+	return fmt.Sprintf("%s/%s:%s", registry, image, slug)
+}
+
+func SlugifyTag(tag string) string {
+	validChars := regexp.MustCompile(`(?i)[^a-zA-Z0-9.\-_]`)
+	temp := validChars.ReplaceAllString(tag, "")
```

Comment:
Do we really want to replace with empty string or do we want to "name" the replacements?
Slugifying `#test` would result in `hashtest`...but it might be messier??

Reply:
Hmmm. Can't see an easy solution to that right now. I guess there won't be that many characters that will actually be replaced so not sure how big of a problem it will be to replace with empty string?